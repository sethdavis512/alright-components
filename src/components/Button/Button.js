import React from 'react';
import PropTypes from 'prop-types';

/** A really great button Button component. */
function Button({ text, type }) {
    return <button type={type}>{text}</button>;
}

Button.propTypes = {
    /** Button Text */
    text: PropTypes.string.isRequired,

    /** Button Type */
    type: PropTypes.string
};

Button.defaultProps = {
    text: 'Button',
    type: 'button'
};

export default Button;
