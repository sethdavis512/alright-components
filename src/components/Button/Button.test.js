import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
    const defaultProps = {
        text: 'Send',
        type: 'submit'
    };

    function renderComponent(props) {
        return shallow(<Button {...defaultProps} {...props} />);
    }

    it('should match submit button snapshot', () => {
        const wrapper = renderComponent();
        expect(wrapper).toMatchSnapshot();
    });

    it('should match regular button snapshot', () => {
        const wrapper = renderComponent({
            text: 'Add',
            type: 'button'
        });
        expect(wrapper).toMatchSnapshot();
    });
});
