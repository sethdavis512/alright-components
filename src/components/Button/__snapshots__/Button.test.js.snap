// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match regular button snapshot 1`] = `
<button
  type="button"
>
  Add
</button>
`;

exports[`Button should match submit button snapshot 1`] = `
<button
  type="submit"
>
  Send
</button>
`;
